google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java

```diff
@@ -265,13 +274,15 @@ public boolean delete(String datasetId, DatasetDeleteOption... options) {
   }
 
   @Override
-  public boolean delete(final DatasetId datasetId, DatasetDeleteOption... options) {
+  public boolean delete(DatasetId datasetId, DatasetDeleteOption... options) {
+    final DatasetId completeDatasetId = datasetId.setProjectId(options().projectId());
```

Comment:
I could see this as possibly causing the same confusion if someone populates the projectId of the DatasetId. (It's certainly less likely that someone would try it, though.)

Maybe raise an error if the projectId is already populated and set to something that doesn't match options().projectId()?